### PR TITLE
Remove aria-hashpopup tooltip prop

### LINKED ISSUE #5300: Please remove aria-haspopup="true" from tooltip markup
If an element has `aria-haspopup="true"`, it means "this element pops up a **menu**".

But a "bx--tooltip__trigger" `div` doesn't pop up a menu, so this is misleading to screen reader users.
(i.e. they will hear "menu button" or something similar, and expect to arrow through menu items; and some screen readers even give a keyboard hint that the user can type down arrow to open the menu, which doesn't make sense here).

Please see the special note in the [ARIA spec for aria-haspopup](https://w3c.github.io/aria/#aria-haspopup):
> NOTE
> A tooltip is not considered to be a popup in this context.

Here is the markup in the [Carbon React Storybook for Tooltip](https://react.carbondesignsystem.com/?path=/story/tooltip--default-bottom):
```html
<div class="bx--tooltip__trigger" role="button" tabindex="0" aria-haspopup="true" aria-labelledby="__carbon-tooltip-trigger_bmbcpfuzeiu" aria-expanded="false">
  <svg>(?)</svg>
</div>
```

Please completely delete `aria-haspopup` from this markup. This attribute is not needed here.

### PR DESCRIPTION
**Pull request by xxxle0**
Closes #5300

+ remove aria-hashpopup prop

#### Changelog

**New**

- {{new thing}}

**Changed**

- {{change thing}}

**Removed**

- {{removed thing}}

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

### CODE REVIEW

**Review by tw15egan — APPROVED** [review]

**Review by aledavila — APPROVED** [review]

**Review by abbeyhrt — APPROVED** [review]

**Comment by carmacleod** [comment]
Thank-you @xxxle0 and reviewers!
@tw15egan How can I tell which version of Carbon react this change will be in?

**Comment by tw15egan** [comment]
Should be included in the next patch release (`10.9.4` / `7.9.4`) 👍

File: packages/components/src/components/tooltip/tooltip.hbs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 <div id="{{noHeading.idSuffix}}-label" class="{{@root.prefix}}--tooltip__label">
   Tooltip label
   <button aria-expanded="false" aria-labelledby="{{noHeading.idSuffix}}-label" data-tooltip-trigger data-tooltip-target="#{{noHeading.idSuffix}}"
-    class="{{@root.prefix}}--tooltip__trigger" aria-controls="{{noHeading.idSuffix}}" aria-haspopup="true">
+    class="{{@root.prefix}}--tooltip__trigger" aria-controls="{{noHeading.idSuffix}}">
```
Comment (elizabethsjudd) [inline packages/components/src/components/tooltip/tooltip.hbs]: @tw15egan why was this removed? This was added for a11y purposes?
